src/h5web/visualizations/heatmap/ColorMapSelector.tsx

```diff
@@ -1,24 +1,56 @@
 import React, { CSSProperties } from 'react';
-import Select, { components } from 'react-select';
+import Select, { components, OptionProps, IndicatorProps } from 'react-select';
 import { FiChevronDown } from 'react-icons/fi';
 import shallow from 'zustand/shallow';
 import { generateCSSLinearGradient } from './utils';
 import { INTERPOLATORS } from './interpolators';
 import { ColorMap, useHeatmapStore } from './store';
 import styles from './HeatmapToolbar.module.css';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const DropdownIndicator = (props: any): JSX.Element => {
+type OptionType = {
+  label: string;
+  value: string;
+};
+
+const DropdownIndicator = (props: IndicatorProps<OptionType>): JSX.Element => {
   return (
     <components.DropdownIndicator {...props}>
       <FiChevronDown />
     </components.DropdownIndicator>
   );
 };
 
+const Option = (props: OptionProps<OptionType>): JSX.Element => {
+  const {
+    data: { label, value },
+    isSelected,
+  } = props;
+
+  return (
+    <components.Option
+      {...props}
+      className={isSelected ? styles.colorMapSelected : styles.colorMapOption}
+    >
+      {label}
+      <div
+        style={{
+          backgroundImage: generateCSSLinearGradient(
+            INTERPOLATORS[value as ColorMap],
+            'right'
+          ),
+          width: '2rem',
+        }}
+      />
+    </components.Option>
+  );
+};
+
 // react-select expects to work with objects of type {label: string, value: string}
 // So I use the color map name as both label and value
-const colorMapOptions = Object.keys(INTERPOLATORS).map(label => ({ label }));
+const colorMapOptions = Object.keys(INTERPOLATORS).map(label => ({
+  label,
+  value: label,
+}));
```

Comment:
`isSelected` does not work properly if there is no `value` field. I then added the field and tried to use `value` and `label` at their designated places even though they are the same.